socorro/submitter/breakpad_submitter_utilities.py

```diff
@@ -1,68 +1,58 @@
-#! /usr/bin/env python
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/
-from configman import Namespace
-from socorro.external.crashstorage_base import CrashStorageBase
 
 import os
 import urllib2
+
+from configman import Namespace
 import poster
+
+from socorro.external.crashstorage_base import CrashStorageBase
+
+
 poster.streaminghttp.register_openers()
 
 
 class BreakpadPOSTDestination(CrashStorageBase):
-    """This a crashstorage derivative that pushes a crash out to a
-    Socorro collector waiting at a url.
+    """This a crashstorage derivative that pushes a crash out to a Socorro collector waiting at a url.
 
     """
     required_config = Namespace()
     required_config.add_option(
-        'url',
-        short_form='u',
-        doc="The url of the Socorro collector to submit to",
-        default="http://127.0.0.1:8882/submit"
+        'urls',
+        doc='One or more urls to submit to separated by commas',
+        default="http://127.0.0.1:8888/submit"
     )
-    required_config.add_option(
-        'echo_response',
-        short_form='e',
-        doc="echo the submission response to stdout",
-        default=False
-    )
-
-    def __init__(self, config, quit_check_callback=None):
-        super(BreakpadPOSTDestination, self).__init__(
-            config,
-            quit_check_callback
-        )
-        self.hang_id_cache = dict()
 
     def save_raw_crash_with_file_dumps(self, raw_crash, dumps, crash_id):
+        """Saves a raw crash by wrapping it up in a multipart/form-data payload and submitting it as an HTTP
+        POST to a collector
+
+        """
+        urls = self.config.urls.split(',')
+
         try:
+            # Create raw crash with file pointers to minidumps for poster to pull from
             for dump_name, dump_pathname in dumps.iteritems():
                 if not dump_name:
                     dump_name = self.config.source.dump_field
                 raw_crash[dump_name] = open(dump_pathname, 'rb')
+
+            # Get uuid for logging purposes
+            uuid = raw_crash.get('uuid', 'NO UUID')
+
+            # Build the payload
             datagen, headers = poster.encode.multipart_encode(raw_crash)
-            request = urllib2.Request(
-                self.config.url,
-                datagen,
-                headers
-            )
-            submission_response = urllib2.urlopen(request).read().strip()
-            try:
+
+            # Submit payload to all urls
+            for url in urls:
+                request = urllib2.Request(url, datagen, headers)
+                submission_response = urllib2.urlopen(request).read().strip()
```

Comment:
Should we maybe try-catch here in case submission fails? That way if it does, we can continue submitting to the other URLs if it's a URL-specific thing.

Reply:
I don't know... This is a transactiony kind of mess. We have that situation whether we add exception handling or not. We can't enforce "submit it to all urls or none of the urls". So if there are two urls, is it better to have it submit to one, but not another?

I think we can be hand-wavey since we don't need to make sure *all* possible things going to *all* -stage environments make it there. We just need to make sure that some things get to each -stage environment so that we know the environment is ok.

Given that, I don't think I want to add exception handling in this case.